ipinfo/cmd_download.go

```diff
@@ -238,3 +268,38 @@ func unzipWrite(file *os.File, data io.Reader) error {
 
 	return nil
 }
+
+func computeSHA256(filepath string) (string, error) {
+	file, err := os.Open(filepath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hasher := sha256.New()
+	if _, err := io.Copy(hasher, file); err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
+}
+
+func fetchChecksums(url string) (*ChecksumResponse, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
```

Comment:
Usually I put this right above the actual readall call instead of after a successful fetch, since we sometimes don't even read the body after a fetch as we may check for HTTP error codes / etc.

So:

```go
resp, err := get()
if err != nil {

}

defer resp.Body.Close()
io.ReadAll()
```